e2e/custom-test-sequencer/testSequencer.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const {getCacheFilePath} = require('../../packages/jest-haste-map/build');
+
+const priorityOrder = [
+  path.resolve('./d.test.js'),
+  path.resolve('./b.test.js'),
+  path.resolve('./c.test.js'),
+];
+const FAIL = 0;
+const SUCCESS = 1;
+
+class TestSequencer {
+  constructor() {
+    this._cache = new Map();
+  }
+  _getCachePath(context) {
+    const {config} = context;
+    return getCacheFilePath(config.cacheDirectory, 'perf-cache-' + config.name);
+  }
+
+  _getCache(test) {
+    const {context} = test;
+    if (!this._cache.has(context) && context.config.cache) {
+      const cachePath = this._getCachePath(context);
+      if (fs.existsSync(cachePath)) {
+        try {
+          this._cache.set(
+            context,
+            JSON.parse(fs.readFileSync(cachePath, 'utf8'))
+          );
+        } catch (e) {}
+      }
+    }
+
+    let cache = this._cache.get(context);
+    if (!cache) {
+      cache = {};
+      this._cache.set(context, cache);
+    }
+
+    return cache;
+  }
+
+  sort(tests) {
```

Comment:
I still don't understand why this im needs to be 100+ lines. Why does it need cache etc? It's just meant to show you can plug in a custom implementation, it doesn't have to be an extensive implementation